Subscribers/Article.php

```diff
@@ -101,10 +101,71 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Backend_Article' => 'extendBackendArticle',
             'Enlight_Controller_Action_PreDispatch_Backend_Article' => 'preBackendArticle',
             'Enlight_Controller_Action_PostDispatch_Frontend_Detail' => 'modifyConnectArticle',
-            'Enlight_Controller_Action_PreDispatch_Frontend_Detail' => 'extendFrontendArticle'
+            'Enlight_Controller_Action_PreDispatch_Frontend_Detail' => 'extendFrontendArticle',
+            'Shopware_Modules_Basket_AddArticle_Start' => 'checkSupplierPluginAvailability'
         ];
     }
 
+    /**
+     * @param \Enlight_Event_EventArgs $args
+     */
+    public function checkSupplierPluginAvailability(\Enlight_Event_EventArgs $args)
+    {
+        $articleDetail = $this->helper->getDetailByNumber($args->getId());
+        $articleDetailId = $articleDetail->getId();
+
+        if ($this->helper->isRemoteArticleDetail($articleDetailId)) {
+            $shopProductId = $this->helper->getShopProductId($articleDetailId);
+            $shopId = $shopProductId->shopId;
+
+            $this->pingRemoteShop($shopId);
+        };
+    }
+
+    /**
+     * @param $shopId
+     */
+    private function pingRemoteShop($shopId)
+    {
+        try {
+            $this->sdk->pingShop($shopId);
```

Comment:
the heck... I can't find this method... can you please point me to the `pingShop` implementation